### PR TITLE
Maintenance checkin for macos 11.6

### PR DESCRIPTION
**Pull request by yungyuc**
Improve the makefile to workaround macos system integrity protection.

Close socket after having the exception to remove the warning.

Force Gmsh helper to use version 2.2 output file format.  This should address issue #236.

### CODE REVIEW

File: solvcon/helper.py
Comment (yungyuc) [inline solvcon/helper.py:299]: This should fix issue #236 .

File: solvcon/connection.py
Comment (yungyuc) [inline solvcon/connection.py:195]: This fixes the many `ResourceWarning`.

File: Makefile
Comment (yungyuc) [inline Makefile:4]: `RUNENV` is for working around the Apple security measures.

File: solvcon/connection.py
Comment (tai271828) [reply solvcon/connection.py:195]: A post-event note:

FYI, I used the following snippet to reduce the warnings:

```
import warnings

warnings.simplefilter("ignore", ResourceWarning)
```

Comment (tai271828) [reply solvcon/connection.py:195]: In my opinion, the current `skt.close()` should be good enough, and maybe better than `warnings.simplefilter("ignore", ResourceWarning)`? Because the latter closes the resources exactly. So I will suggest let us move forward.

**Review by tai271828 — APPROVED** [review]
Thanks for improving this! I can't make it without macOS. A comment regarding the resource warning for your reference.

**Comment by tai271828** [comment]
Uh, interesting, I don't have merge permission : )

**Comment by tai271828** [comment]
So far it's ok for me to just have review permission. I am more than happy to review pull requests and let someone else make the final decision of landing the code.

File: solvcon/connection.py
Comment (yungyuc) [reply solvcon/connection.py:195]: Yes, I think not closing the socket is the root cause of the warning and we should resolve it.

**Comment by yungyuc** [comment]
> So far it's ok for me to just have review permission. I am more than happy to review pull requests and let someone else make the final decision of landing the code.

I think it's an overlook.  We made too little progress with this repository so I forgot to update the permission for you.  Now you get write.  But I'll just merge this one.

File: Makefile
Comment (tai271828) [inline Makefile:11]: This trick raise a regression  of `make install`. See https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_node/make_79.html `If the file name contains no slash, the directory part is the string `./'.`. That being said, the `dir` evaluated value of `PREFIX ?= $(realpath $(dir ${PYTHON})/../)` is `./ ./some-real-path`. This makes `PREFIX` in `.PHONY install` session is unexpected, and `cp libmarch.*.so ${PYTHON_LIBRARY_DIR}` will fail.

Comment (yungyuc) [reply Makefile:11]: I am not sure I can interpret the literals  correctly because the formatting seems garbled.

Does #240 address this issue?

Comment (tai271828) [reply Makefile:11]: Yeeeeeeeeeeeeeeeeeeeeeeeeeees!!!!!!

```
...................................................................................................................................................................................................................................
................................................................................................................................................................SSS                                                                
----------------------------------------------------------------------                                                                                                                                                             
Ran 390 tests in 3.998s                                                                                                                                                                                                            
                                                                                                                                                                                                                                   
OK (SKIP=3)                                                                                                                                                                                                                        
.                                                                                                                                                                                                                                  
----------------------------------------------------------------------                                                                                                                                                             
Ran 1 test in 0.282s                                                                                                                                                                                                               
                                                                                                                                                                                                                                   
OK 
```

**Pull request merged**
